src/component/Access/access-context.js

```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+
+const AccessContext = React.createContext()
+
+const AccessProvider = ({store, children}) => {
+  const hasAccess = (permission, project) => {
+    const permissions = store.getState().user.get('permissions');
+
+    return permissions.some(p => {
+      if(p.permission === 'ADMIN') {
+        return true
+      }
```

Comment:
Does this method need a check to match permission and return true if project is undefined?

Reply:
If the users has the ADMIN permission we do not need to check the project as it will take precedence.

Reply:
Yes. But a user with the Editor role should be able to create and edit addons (at least according to the current description). The way I'm reading this is that we return early if they have ADMIN privileges, and if not we do a check that matches the incoming permission with the user permission and permission project with incoming project. Since addons do not have projects, aren't we going to miss this use case here? Or am I missing something?

Reply:
Yes you are. The ADMIN permission is a special permission only granted if you belong to the admin role. This permission automatically wins over all other permissions. 

E.g. You need the ADMIN permission to manage users. We might introduce a special permission later for CREATE/UPDATE/DELETE users, but we have not yet. 

Having the ADMIN permission also simplifies the access to all project resources, as this permission will not require you to have a specific permission for for the given project.